tools/websoc-scraper-v2/index.ts

```diff
@@ -414,44 +420,41 @@ async function scrape(name: string, term: Term) {
 /**
  * The entry point of the program.
  */
-(async function main() {
+async function main() {
   try {
-    logger.info("websoc-scraper-v2 daemon starting");
-    let now = new Date();
-    let termsInScope = await getTermsToScrape(now);
-    for (const [name, term] of Object.entries(termsInScope))
-      await scrape(name, term);
-    for (;;) {
-      const curr = new Date();
-      // Check the available terms every day and scrape all of them once.
-      if (curr.getDate() !== now.getDate()) {
-        termsInScope = await getTermsToScrape(curr);
-        for (const [name, term] of Object.entries(termsInScope))
-          await scrape(name, term);
-      }
-      now = curr;
-      // Check the database for terms to scrape on demand.
-      const websocTerms = await prisma.websocTerm.findMany({
-        where: { timestamp: { gte: now } },
-        select: { year: true, quarter: true },
-      });
-      const termsOnDemandEntries = websocTerms.map(
-        (x) => [`${x.year} ${x.quarter}`, x] as TermEntry
-      );
-      const termsOnDemand = Object.fromEntries(termsOnDemandEntries);
-      let scraped = false;
-      for (const [name, term] of Object.entries(termsOnDemand)) {
-        if (!(name in termsInScope)) {
-          logger.info(`Term ${name} requested but not in scope, skipping`);
-          continue;
-        }
-        scraped = true;
+    logger.info("websoc-scraper-v2 starting");
+    const prev = new Date(
+      existsSync("/tmp/websoc_scraper_v2")
+        ? parseInt(readFileSync("/tmp/websoc_scraper_v2").toString())
+        : Date.now() - 24 * 60 * 60 * 1000
+    );
+    const termsInScope = await getTermsToScrape(prev);
+    const curr = new Date();
+    // If it is a new day, scrape all terms.
+    if (curr.getDate() !== prev.getDate()) {
+      for (const [name, term] of Object.entries(termsInScope))
         await scrape(name, term);
+    }
+    // Check the database for terms to scrape on demand.
+    const websocTerms = await prisma.websocTerm.findMany({
+      where: { timestamp: { gte: curr } },
+      select: { year: true, quarter: true },
+    });
+    const termsOnDemandEntries = websocTerms.map(
+      (x) => [`${x.year} ${x.quarter}`, x] as TermEntry
+    );
```

Comment:
```suggestion
    const termsOnDemandEntries: TermEntry[] = websocTerms
      .map((x) => [`${x.year} ${x.quarter}`, x])
      .filter(([name]) => {
        if (!(name in termsInScope)) {
          logger.info(`Term ${name} requested but not in scope, skipping`);
          return false;
        }
        return true;
    })
```
I think you could actually define the type upfront and then the returned arrays are forced to comply. Also, I think you can remove the `continue` block in the code below just by filtering out the invalid terms early.